packages/bull-vault/test/integration-test/AuctionBullTestFork.t.sol

```diff
@@ -2023,6 +2035,133 @@ contract AuctionBullTestFork is Test {
         assertEq(squeethInCrab.add(wPowerPerpAmountToTrade), squeethInCrabAfter);
     }
 
+    function testFullRebalanceReduceWethInEuler() public {
+        currentDebt = IEulerDToken(dToken).balanceOf(address(bullStrategy));
+        currentWethInLeverage = IEulerEToken(eToken).balanceOfUnderlying(address(bullStrategy));
+        console.log('currentWethInLeverage',currentWethInLeverage);    
+        {
+            // move prices up
+            vm.startPrank(user1);
+            IWETH9(weth).deposit{value: 200e18}();
+            IERC20(weth).approve(address(swapRouter), 200e18);
+            swapRouter.exactInputSingle(
+                ISwapRouter.ExactInputSingleParams({
+                    tokenIn: weth,
+                    tokenOut: wPowerPerp,
+                    fee: uint24(3000),
+                    recipient: msg.sender,
+                    deadline: block.timestamp,
+                    amountIn: 200e18,
+                    amountOutMinimum: 0,
+                    sqrtPriceLimitX96: 0
+                })
+            );
+            vm.stopPrank();
+            vm.warp(block.timestamp + 1000);
+        }
+
+        uint256 ethUsdPrice = UniOracle._getTwap(
+            controller.ethQuoteCurrencyPool(),
+            controller.weth(),
+            controller.quoteCurrency(),
+            TWAP,
+            false
+        );
+
+        (targetWethInLeverage, targetDebt) = _calcTargetCollateralAndDebtInLeverage();
+        console.log('targetWethInLeverage',targetWethInLeverage);    
+
+        (uint256 crabAmount, bool isDepositingInCrab) = _calcCrabAmountToTrade(
+            currentWethInLeverage, currentDebt, targetWethInLeverage, targetDebt, ethUsdPrice
+        );
+        console.log('isDepositingInCrab',isDepositingInCrab);    
+
+        assertEq(isDepositingInCrab, true);
```

Comment:
I think would be good to do more asserts like this to confirm tests are doing what we want.

ie could assertEq(isDepositingInEuler) and assertEq(isBorrowingFromEuler) and have it for all tests to make sure they are doing what we expect and to make sure we have them all